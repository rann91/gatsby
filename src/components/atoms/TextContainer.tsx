/* @jsx jsx */
import { PropsWithChildren } from 'react'
import { Box, jsx, SxProps } from 'theme-ui'
import { rem } from '../../gatsby-plugin-theme-ui'

type Props = SxProps & PropsWithChildren<{}>

const TextContainer = ({ children, ...rest }: Props) => (
  <Box
    data-testid="text-container"
    sx={{ margin: 'auto', maxWidth: rem(940), textAlign: 'center' }}
    {...rest}>
    {children}
  </Box>
)

export default TextContainer
