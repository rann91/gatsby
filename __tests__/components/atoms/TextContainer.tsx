import { render } from '@testing-library/react'
import React from 'react'
import TextContainer from '../../../src/components/atoms/TextContainer'

describe('TextContainer', () => {
  it('renders correctly', () => {
    const { container } = render(<TextContainer />)

    expect(container).toMatchSnapshot()
  })
})
