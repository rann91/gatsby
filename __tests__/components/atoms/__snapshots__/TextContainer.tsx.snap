// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextContainer renders correctly 1`] = `
<div>
  <div
    class="css-ycxng9-TextContainer"
    data-testid="text-container"
  />
</div>
`;
